src/extensions/Markdown.js

```diff
@@ -78,19 +78,27 @@ const Markdown = Extension.create({
 			new Plugin({
 				key: new PluginKey('pasteEventHandler'),
 				props: {
+					handleDOMEvents: {
+						mouseup(_, event) {
+							shiftKey = event.shiftKey
+							return false
+						},
+					},
 					handleKeyDown(_, event) {
 						shiftKey = event.shiftKey
 						return false
 					},
 					clipboardTextParser(str, $context, _, view) {
-						if (shiftKey) {
-							return
-						}
+						const parser = DOMParser.fromSchema(view.state.schema)
 						const doc = document.cloneNode(false)
 						const dom = doc.createElement('div')
-						dom.innerHTML = markdownit.render(str)
+						if (shiftKey) {
+							// Treat single newlines as linebreaks and double newlines as paragraph breaks when pasting as plaintext
+							dom.innerHTML = '<p>' + str.replaceAll('\n', '<br />').replaceAll('<br /><br />', '</p><p>') + '</p>'
```

Comment:
I'm wondering if we should ignore trailing new lines.
If i paste an entire line including the trailing new line I would not expect to have a hard break at the end of the inserted paragraph.

Reply:
I guess the same goes for leading newline chars.

Reply:
I tried it myself... works good enough. Plus this seems like a super rare scenario given that this paste only happens on shift+ctrl+v.
So let's stick to the most generic approach.